api_pos_integration.md

```diff
@@ -0,0 +1,277 @@
+```
+snabble:752dd716-ec0c-11e8-8528-68f7286a148f
+```
+
+The string after the colon is the identifier of the checkout process.
+
+To build the URL of the process the base URL of the used snabble
+environment (see [General API access:
+Environments](api_general.md#environments)), the identifier of the
+project and the identifier of the checkout is needed. This is the URL
+template:
+
+```
+<base URL>/<project id>/pos/checkout/id/<checkout id>
+```
+ie. `https://api.snabble.io/test-project-12345/pos/checkout/id/752dd716-ec0c-11e8-8528-68f7286a148f`.
+
+After the PoS has accessed the identifier of the checkout process, it
+should update the state of the checkout process to communicate that it
+started to process it. Additionally it might identify itself by
+sending an unique identifier. The service responds with the complete
+updated checkout process. This allows the PoS to process the
+checkout. The call to the service might look like:
+
+```
+PATCH /project/pos/checkout/id/752dd716-ec0c-11e8-8528-68f7286a148f HTTP/1.1
+Host: api.snabble.io
+Content-Type: application/merge-patch+json
+Client-Token: ...
+
+{
+  "state": "processing",
+  "processedBy": "pos-1"
+}
+```
+The service responds with:
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "lineItems" : [
+    {
+      "scannedCode": "0000000000001",
+      "totalPrice" : 798,
+      "amount" : 2,
+      "price" : 399,
+      "taxRate" : 19,
+      "sku" : "1"
+    },
+    {
+      "scannedCode": "0000000000002"
+      "price" : 1099,
+      "totalPrice" : 1099,
+      "amount" : 1,
+      "taxRate" : 19
+    }
+  ],
+  "tax" : {
+    "19" : 7673
+  },
+  "taxNet" : {
+    "19" : 40382
+  },
+  "netPrice" : 40382,
+  "price" : 48055,
+  "state": "processing",
+  "processedBy": "pos-1",
+  "loyaltyCard": "12345678"
+}
+```
+
+After successfully processing the checkout the PoS should update the
+checkout process. First, it should set the state to
+`successful`. Second, it should update the line items and the price to
+reflect the performed checkout.
+
+```
+PATCH /project/pos/checkout/id/752dd716-ec0c-11e8-8528-68f7286a148f HTTP/1.1
+Host: api.snabble.io
+Content-Type: application/merge-patch+json
+Client-Token: ...
+
+{
+  "lineItems" : [
+    {
+      "scannedCode": "0000000000001",
+      "totalPrice" : 798,
+      "amount" : 2,
+      "price" : 399,
+      "taxRate" : 19,
+      "sku" : "1"
+    },
+    {
+      "scannedCode": "0000000000002"
+      "price" : 1099,
+      "totalPrice" : 1099,
+      "amount" : 1,
+      "taxRate" : 19
+    }
+  ],
+  "tax" : {
+    "19" : 7673
+  },
+  "taxNet" : {
+    "19" : 40382
+  },
+  "netPrice" : 40382,
+  "price" : 48055,
+  "state": "successful",
+  "loyaltyCard": "12345678"
+}
+```
+
+If the checkout could not be performed, the PoS should update the
+checkout process and set its state to `failed`. Optionally the PoS can
+provide an informal `failedReason` message, which describes the
+conditions of the failure.
```

Comment:
"which contains the reasons of the failure" Die Property heißt ja auch failedReason